relayer/strategies.go

```diff
@@ -31,42 +30,55 @@ const (
 func StartRelayer(
 	ctx context.Context,
 	log *zap.Logger,
-	src, dst *Chain,
-	filter ChannelFilter,
+	chains map[string]*Chain,
+	paths []NamedPath,
 	maxTxSize, maxMsgLength uint64,
 	memo string,
 	processorType string,
 	initialBlockHistory uint64,
-	pathName string,
 	metrics *processor.PrometheusMetrics,
 ) chan error {
 	errorChan := make(chan error, 1)
 
 	switch processorType {
 	case ProcessorEvents:
-		var filterSrc, filterDst []processor.ChainChannelKey
+		chainProcessors := make([]processor.ChainProcessor, 0, len(chains))
 
-		for _, ch := range filter.ChannelList {
-			ruleSrc := processor.ChainChannelKey{ChainID: src.ChainProvider.ChainId(), ChannelKey: processor.ChannelKey{ChannelID: ch}}
-			ruleDst := processor.ChainChannelKey{CounterpartyChainID: src.ChainProvider.ChainId(), ChannelKey: processor.ChannelKey{CounterpartyChannelID: ch}}
-			filterSrc = append(filterSrc, ruleSrc)
-			filterDst = append(filterDst, ruleDst)
+		for _, chain := range chains {
+			chainProcessors = append(chainProcessors, chain.chainProcessor(log, metrics))
 		}
-		paths := []path{{
-			src: pathChain{
-				provider: src.ChainProvider,
-				pathEnd:  processor.NewPathEnd(pathName, src.ChainProvider.ChainId(), src.ClientID(), filter.Rule, filterSrc),
-			},
-			dst: pathChain{
-				provider: dst.ChainProvider,
-				pathEnd:  processor.NewPathEnd(pathName, dst.ChainProvider.ChainId(), dst.ClientID(), filter.Rule, filterDst),
-			},
-		}}
-
-		go relayerStartEventProcessor(ctx, log, paths, initialBlockHistory, maxTxSize, maxMsgLength, memo, errorChan, metrics)
+
+		ePaths := make([]path, len(paths))
+		for _, np := range paths {
+			pathName := np.Name
+			p := np.Path
+
+			filter := p.Filter
+			var filterSrc, filterDst []processor.ChainChannelKey
+
+			for _, ch := range filter.ChannelList {
+				ruleSrc := processor.ChainChannelKey{ChainID: p.Src.ChainID, ChannelKey: processor.ChannelKey{ChannelID: ch}}
+				ruleDst := processor.ChainChannelKey{CounterpartyChainID: p.Src.ChainID, ChannelKey: processor.ChannelKey{CounterpartyChannelID: ch}}
+				filterSrc = append(filterSrc, ruleSrc)
+				filterDst = append(filterDst, ruleDst)
+			}
+			ePaths = append(ePaths, path{
+				src: processor.NewPathEnd(pathName, p.Src.ChainID, p.Src.ClientID, filter.Rule, filterSrc),
+				dst: processor.NewPathEnd(pathName, p.Dst.ChainID, p.Dst.ClientID, filter.Rule, filterDst),
+			})
+		}
+
+		go relayerStartEventProcessor(ctx, log, chainProcessors, ePaths, initialBlockHistory, maxTxSize, maxMsgLength, memo, errorChan, metrics)
 		return errorChan
 	case ProcessorLegacy:
-		go relayerMainLoop(ctx, log, src, dst, filter, maxTxSize, maxMsgLength, memo, errorChan)
+		if len(paths) != 1 {
+			panic(fmt.Errorf("only one path supported for legacy processor. pass `-p events` for multiple paths"))
```

Comment:
#989 switches to the event processor as the default relayer behavior so maybe this error message needs to be changed slightly? I would imagine at that point explicitly using `-p events` is no longer needed and something like `-p l` would be used to switch to the legacy processor

Reply:
will include this in #989